### PR TITLE
Add the ability to choose the I2C master id

### PR DESCRIPTION
**Pull request by 0r31**
Hi,

### Description

This PR adds the ability to choose the I2C master id from the vendor **Version.h** file.
This is LPC176x specific and a change in the related framewok is mandatory to make it work.
That's why I created a PR on the fmk too : https://github.com/p3p/pio-framework-arduino-lpc176x/pull/30

In waiting for the fmk PR to be integrated/released, this code change is useless but can be used in manually modify the fmk sources grabbed by platformio.

### Notes

The **USEDI2CDEV_M** identifier is currently defined (and set to 1) twice in Marlin sources: [here](https://github.com/MarlinFirmware/Marlin/blob/ccdeb3f2cd291e050f6a5e6f503455e697f893e8/Marlin/src/HAL/HAL_LPC1768/u8g/LCD_I2C_routines.cpp#L72) and [here](https://github.com/MarlinFirmware/Marlin/blob/ccdeb3f2cd291e050f6a5e6f503455e697f893e8/Marlin/src/HAL/HAL_LPC1768/include/digipot_mcp4451_I2C_routines.h#L27).

As it is now defaulted to 1 in the main **Version.h** file, i think these two lines can be removed but i didn't include it in the PR. Could you please double-check? I would enhance the PR with these lines removal if needed.

### The future

32-bits mcus seem to have many available I2C buses. For instance, the LPC1769 have three. As far as i understand, it should be nice to be able to instantiate them independantly. It might mean to do something in at least the LPC176x fmk similar to what was originally done for HardwareSerial in the standard Arduino library (HardwareSerial0, HardwareSerial0, ... ---> Wire0, Wire0, ...).
I am aware this might be a big work on both Marlin and the LPC176x fmk. Just my two cents.

Regards,
Orel

### CODE REVIEW

**Comment by 0r31** [comment]
Hi @thinkyhead ,

First of all, many thanks for your advice (no platform-specifics on Version files, i was not aware of that) and for your commit.

But i guess i have to add explanations on this pull request. Actually, its first goal was to allowing users to define the I2C master id to be used without modifying the Marlin core code (so in using any configurable file). That's why i put it in Version files.

From your commit, it seems to be not possible anymore.

One one hand, one has to modify either [LCD_I2C_routines.cpp](https://github.com/MarlinFirmware/Marlin/blob/d035bfb1a97357c27b04e190c7e2a844df7dbd0f/Marlin/src/HAL/HAL_LPC1768/u8g/LCD_I2C_routines.cpp#L72) or [digipot_mcp4451_I2C_routines.h](https://github.com/MarlinFirmware/Marlin/blob/d035bfb1a97357c27b04e190c7e2a844df7dbd0f/Marlin/src/HAL/HAL_LPC1768/include/digipot_mcp4451_I2C_routines.h#L27).
On the other hand, there is no way to define the I2C master id except in hardcoding it before including Wire (e.g. [here](https://github.com/MarlinFirmware/Marlin/blob/ccdeb3f2cd291e050f6a5e6f503455e697f893e8/Marlin/src/feature/dac/dac_mcp4728.h#L30)) when directly using the [LPC176x wire library](https://github.com/p3p/pio-framework-arduino-lpc176x/blob/master/cores/arduino/Wire.cpp) + the [PR i sent](https://github.com/p3p/pio-framework-arduino-lpc176x/pull/30) on it.

Consequently, i guess a good compromise can be to add a new identifier (**I2C_MASTER_ID** for instance) in _Configuration.h_ and use it to define USEDI2CDEV_M when needed.

To sum up, the main goal is to allow LPC176x boards to select the I2C master id regarding existing code and I2C devices.

What do you think?

Regards,
Orel

Additional Note: I noticed all the stuff you gathered (i2c initialization) is already done in the [LPC176x Wire library begin() function](https://github.com/p3p/pio-framework-arduino-lpc176x/blob/91b9f7cae8e1a1de04e8a2a9246bb36dd9e4c5dc/cores/arduino/Wire.cpp#L63). I tend to say _LCD_I2C_routines.cpp_ or _digipot_mcp4451_I2C_routines.h_ could directly use it. There might need one missing point : the **clock_option** argument.

**Comment by thinkyhead** [comment]
Anything that sees `_Version.h` would be including `inc/MarlinConfigPre.h` so the correct place to provide user-configurable options is `Configuration.h` or `Configuration_adv.h`. With my changes if you add an override to either one of those files, it will be seen by all the files that would have seen it in `_Version.h`.

**Comment by 0r31** [comment]
Hi @thinkyhead ,

Understood. I will create another PR with config file modifications.

Regards,
Orel

**Comment by ellensp** [comment]
Can no longer enable U8GLIB_SSD1306 on LPC1768/9 boards after this was merged into bugfix. 
The error: "Marlin/src/HAL/HAL_LPC1768/u8g/LCD_I2C_routines.cpp : 100 : undefined reference to `configure_i2c(unsigned char)'"

**Comment by 0r31** [comment]
Hi @ellensp ,

It seems using `extern C` in **i2c_util.h** fixes this issue.

```C
#ifdef __cplusplus
  extern "C" {
#endif

void configure_i2c(const uint8_t clock_option);

#ifdef __cplusplus
  }
#endif
```

_I also checked if it still works with DIGIPOT_I2C enabled._

But i guess we have to check with @thinkyhead if it is correct.
If so, i will add a commit to #16636 .

Regards,
Orel

**Comment by ellensp** [comment]
Confirmed, this compiles and works.

**Comment by 0r31** [comment]
Hi @ellensp ,

When you say "this works", do you mean the screen is usable? (I don't have any to make tests).

Regards,
Orel

**Comment by ellensp** [comment]
As discussed. Seems to work. I get boot screen and status screen.  I only have test setup. A re-arm and a DDS1306, no buttons or encoders to test beyond this.

**Comment by Bob-the-Kuhn** [comment]
This PR replaced **digipot_mcp4451_init()** with **configure_i2c()**.

Where is the **configure_i2c()** routine?  Without it the MKS Sbase won't compile.

---

EDIT

Looks like the call should be **configure_i2c(clock_option)** but I don't see where clock_option is set to any value.

**Comment by Bob-the-Kuhn** [comment]
PR #16662 has been submitted to fix the MKS Sbase problem.
